ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -802,7 +801,20 @@ let on_notification server (notification : Client_notification.t) :
        after receiving this notification *)
     let configuration = Configuration.update state.configuration req in
     Fiber.return { state with configuration }
-  | DidSaveTextDocument _
+  | DidSaveTextDocument { textDocument = { uri }; _ } -> (
+    let open Fiber.O in
+    let state = Server.state server in
+    match Document_store.get_opt state.store uri with
+    | None ->
+      ( Log.log ~section:"on receive DidSaveTextDocument" @@ fun () ->
+        Log.msg "saved document is not in the store" [] );
+      Fiber.return state
+    | Some doc ->
+      (* we need new pipeline to get updated errors *)
+      let* doc = Document.doc_with_new_merlin_pipeline doc in
```

Comment:
This code is too similar to how we handle `TextDocumentDidChange`. In fact it's exactly the same if you take the list of changes to be empty. I suggest to combine the two and remove the `doc_with_new_merlin_pipeline` function altogether.